spec/Converter/ShippingAdjustmentsToLineItemsConverterSpec.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\InvoicingPlugin\Converter;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Core\Model\ShipmentInterface;
+use Sylius\InvoicingPlugin\Converter\LineItemsConverterInterface;
+use Sylius\InvoicingPlugin\Entity\LineItem;
+use Sylius\InvoicingPlugin\Provider\TaxRatePercentageProviderInterface;
+
+final class ShippingAdjustmentsToLineItemsConverterSpec extends ObjectBehavior
+{
+    function let(TaxRatePercentageProviderInterface $taxRatePercentageProvider): void
+    {
+        $this->beConstructedWith($taxRatePercentageProvider);
+    }
+
+    function it_implements_line_items_converter_interface(): void
+    {
+        $this->shouldImplement(LineItemsConverterInterface::class);
+    }
+
+    function it_extracts_line_items_from_orders_shipping_adjusments(
```

Comment:
We could cover a few more cases here, like lack of tax adjustment or having two of them.